app/lib/models.dart

```diff
@@ -188,3 +208,55 @@ String niceUrl(String url) {
   }
   return url;
 }
+
+/// Sorts [versions] according to the semantic versioning specification.
+///
+/// If [pubSorting] is `true` then pub's priorization ordering is used, which
+/// will rank pre-release versions lower than stable versions (e.g. it will
+/// order "0.9.0-dev.1 < 0.8.0").  Otherwise it will use semantic version
+/// sorting (e.g. it will order "0.8.0 < 0.9.0-dev.1").
+void sortPackageVersionsDesc(List<PackageVersion> versions,
+    {bool decreasing: true, bool pubSorting: true}) {
+  versions.sort(
+      (PackageVersion a, PackageVersion b) => _compareSemanticVersionsDesc(
+            a.semanticVersion,
+            b.semanticVersion,
+            decreasing: decreasing,
+            pubSorting: pubSorting,
+          ));
+}
+
+/// Compares two versions according to the semantic versioning specification.
+///
+/// If [pubSorting] is `true` then pub's priorization ordering is used, which
+/// will rank pre-release versions lower than stable versions (e.g. it will
+/// order "0.9.0-dev.1 < 0.8.0").  Otherwise it will use semantic version
+/// sorting (e.g. it will order "0.8.0 < 0.9.0-dev.1").
+int _compareSemanticVersionsDesc(Version a, Version b,
+    {bool decreasing: true, bool pubSorting: true}) {
```

Comment:
This is a private method and only called from two places, I would make the named parameters, required positional ones (both call sites know the information).